addons/website_sale_stock/models/website.py

```diff
@@ -0,0 +1,11 @@
+# -*- coding: utf-8 -*-
+from openerp import api, fields, models
+
+class Website(models.Model):
+    _inherit = 'website'
+
+    stock_warning_active =  fields.Boolean(compute="_compute_options", string='Out of Stock Warning')
```

Comment:
Don't do that ! Execute your `_website_product_id_change` even if the view is not activated.